### PR TITLE
Update language server to use new linter for hints

### PR DESCRIPTION
**Pull request by dsainati1**
Closes https://github.com/onflow/cadence/issues/1269

This updates the language servers hint architecture to use the new linter instead of Cadence's hints, which have been removed. 

______

- [X] Targeted PR against `master` branch
- [X] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [X] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [X] Updated relevant documentation 
- [X] Re-reviewed `Files changed` in the Github PR explorer
- [X] Added appropriate labels

### CODE REVIEW

File: languageserver/server/server.go
Comment (turbolent) [inline languageserver/server/server.go:3146]: Duplicated here and below. Maybe refactor out the assignment to `protocolDiagnostic` after the switch. The only difference in the cases is the message

**Review by turbolent — COMMENTED** [review]
Nice!

File: languageserver/server/server.go
Comment (turbolent) [inline languageserver/server/server.go:1987]: Might be good to "cache" the result of `maps.Values(linter.Analyzers)`, maybe even on the linter side, e.g. through a getter function that memoizes

File: languageserver/server/server.go
Comment (turbolent) [inline languageserver/server/server.go:1977]: Is this lock required because the analysis below runs concurrently?

Comment (dsainati1) [reply languageserver/server/server.go:1977]: Yeah, it looks like it based on the implementation in `tools/analysis/program.go` as well as this being how the linter itself implements its report function.

Comment (turbolent) [reply languageserver/server/server.go:1979]: Probably not a big deal, but there might be some contention here. We might want to change `analysis.Program.Run` to not accept a single reporting function that is used for all diagnostics of all given analyzers, but report diagnostics for each analyzer separately, like Go does: https://github.com/golang/tools/blob/d6c099e3c1a39214a49a5b7c5c5faa604650e581/go/analysis/internal/checker/checker.go#L687.
We can do that in separate PR(s) though.

File: languageserver/server/server.go
Comment (turbolent) [inline languageserver/server/server.go:3139]:
```suggestion
	}
	
	if message == "" {
		message = linterDiagnostic.Message
	}
```

**Review by turbolent — APPROVED** [review]

**Comment by psiemens** [comment]
Is this good to merge? It's blocking https://github.com/onflow/cadence/pull/1775, which we need for some CLI work. (cc @sukantoraymond).

Thanks for making this change BTW! The new linter package looks awesome.

**Comment by dsainati1** [comment]
This is blocked on https://github.com/onflow/cadence-lint/pull/3, but as soon as that merges I will merge this as well

**Pull request merged**
